### PR TITLE
Approve staking

### PR DESCRIPTION
**Pull request by vujevits**
This branch is based off of `improve-swap-ux`, to compare with that see:
https://github.com/tenderize/tender-app/compare/94c8f75015586099437eb8d51c38acb3d1419001...cc18568daee0f5cfcccda6ec6c831d518a65c68f

- add same component for approval in staking section
- moved tokenApproval to a custom hook

### CODE REVIEW

**Review by kyriediculous — COMMENTED** [review]
Looks good to me but still have to check the preview, I changed the base to `improve-swap-ux`

File: packages/app/src/components/actions/deposit.tsx [hunk 1]
```diff
@@ -59,39 +54,21 @@ const Deposit: FC<Props> = ({ name, symbol, tokenBalance, tokenAllowance }) => {
...
-        {!depositInput || BigNumber.from(tokenAllowance).gte(utils.parseEther(depositInput || "0")) ? (
+        <div className="d-grid gap-2">
+          <ApproveToken
+            symbol={symbol}
+            spender={addresses[name].controller}
+            tokenAddress={contracts[name].token}
+            hasAllowance={isTokenApproved}
+          />
           <Button
             disabled={!depositInput || depositInput.toString() === "0" || depositTx.status === "Mining"}
```
Comment (kyriediculous) [inline packages/app/src/components/actions/deposit.tsx:65]: Need to also add `!isTokenApproved` to the disabled conditions 

image

**Review by kyriediculous — CHANGES_REQUESTED** [review]
Just one small thing that I found

File: packages/app/src/components/actions/deposit.tsx (hunk 1 above)
Comment (vujevits) [reply packages/app/src/components/actions/deposit.tsx:65]: 👍

**Review by kyriediculous — APPROVED** [review]

**Pull request merged**